src/lib/core/trigger.cc

```diff
@@ -50,3 +62,66 @@ trigger_run_reverse(struct rlist *list, void *event)
 			return -1;
 	return 0;
 }
+
+void
+trigger_fiber_run(struct rlist *list, void *event, double timeout)
+{
+	struct trigger *trigger, *tmp;
+	unsigned num = 0;
+
+	/*
+	 * Calculating the total number of triggers.
+	 */
+	rlist_foreach_entry(trigger, list, link)
+		num++;
+
+	RegionGuard(&fiber()->gc);
```

Comment:
You have single exit point, so not sure that using c++ legacy is suitable in this case...